presto-main/src/main/java/com/facebook/presto/execution/SqlQueryQueueManager.java

```diff
@@ -162,7 +162,8 @@ public boolean submit(QueryExecution queryExecution, Executor executor, SqlQuery
             }
         }
 
-        return queues.get(0).enqueue(createQueuedExecution(queryExecution, queues.subList(1, queues.size()), executor, stats));
+        queues.get(0).enqueue(createQueuedExecution(queryExecution, queues.subList(1, queues.size()), executor, stats));
+        return true;
```

Comment:
I have added a test case (see testSqlQueryQueueManagerWithTooManyQueriesScheduled) where I pass more queries than queue can accept queries fails as it is expected.

Queue won't accept unlimited number of queries as it only accepts queries to which it is able to give permits. See linies 157-163 in this file.

Or I missed something, in that case I would be grateful for detailed explanation.